tests/js/server/aql/aql-index-choice.js

```diff
@@ -0,0 +1,513 @@
+        `FOR doc IN ${cn} FILTER doc.uid >= 1234 SORT doc.dt RETURN doc`
+      ].forEach((q) => {
+        let nodes = AQL_EXPLAIN(q).plan.nodes;
+        assertEqual(1, nodes.filter((n) => n.type === 'IndexNode').length);
+        assertEqual(1, nodes.filter((n) => n.type === 'SortNode').length);
+        let indexNode = nodes.filter((n) => n.type === 'IndexNode')[0];
+        assertEqual([], indexNode.projections);
+        assertFalse(indexNode.indexCoversProjections);
+        assertEqual(1, indexNode.indexes.length);
+        let index = indexNode.indexes[0];
+        assertEqual("persistent", index.type);
+        assertEqual(["uid", "dt"], index.fields);
+      });
+    },
+    
+    testMultipleIndexesMultipleFiltersAndSort: function() {
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid"] });
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid", "dt"] });
+      
+      [
+        `FOR doc IN ${cn} FILTER doc.uid == 1234 && doc.dt == 1234 SORT doc.dt RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid == 1234 && doc.dt >= 1234 SORT doc.dt RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid == 1234 && doc.dt <= 1234 SORT doc.dt RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid == 1234 && doc.dt >= 1234 && doc.dt < 9999 SORT doc.dt RETURN doc`,
+      ].forEach((q) => {
+        let nodes = AQL_EXPLAIN(q).plan.nodes;
+        assertEqual(1, nodes.filter((n) => n.type === 'IndexNode').length);
+        assertEqual(0, nodes.filter((n) => n.type === 'SortNode').length);
+        let indexNode = nodes.filter((n) => n.type === 'IndexNode')[0];
+        assertEqual([], indexNode.projections);
+        assertFalse(indexNode.indexCoversProjections);
+        assertEqual(1, indexNode.indexes.length);
+        let index = indexNode.indexes[0];
+        assertEqual("persistent", index.type);
+        assertEqual(["uid", "dt"], index.fields);
+      });
+
+      [
+        `FOR doc IN ${cn} FILTER doc.uid > 1234 && doc.dt == 1234 SORT doc.dt RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid >= 1234 && doc.dt >= 1234 SORT doc.dt RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid >= 1234 && doc.dt <= 1234 SORT doc.dt RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid >= 1234 && doc.dt >= 1234 && doc.dt < 9999 SORT doc.dt RETURN doc`
+      ].forEach((q) => {
+        let nodes = AQL_EXPLAIN(q).plan.nodes;
+        assertEqual(1, nodes.filter((n) => n.type === 'IndexNode').length);
+        assertEqual(1, nodes.filter((n) => n.type === 'SortNode').length);
+        let indexNode = nodes.filter((n) => n.type === 'IndexNode')[0];
+        assertEqual([], indexNode.projections);
+        assertFalse(indexNode.indexCoversProjections);
+        assertEqual(1, indexNode.indexes.length);
+        let index = indexNode.indexes[0];
+        assertEqual("persistent", index.type);
+        assertEqual(["uid", "dt"], index.fields);
+      });
+    },
+  };
+}
```

Comment:
How about a test where we have two indexes, one one `uid` and one on `dt`; in the query we filter on `uid` and sort on `dt` or vice versa. Which filter should be picked? The one that covers the filter or the one that covers the sort?

Reply:
Added